src/Particle/LongRange/KContainer.h

```diff
@@ -26,16 +27,57 @@ namespace qmcplusplus
  * It generates a set of k-points that are unit-translations of the reciprocal-space
  * cell. K-points are generated within a spherical cutoff set by the supercell
  */
-class KContainer : public QMCTraits
+template<typename REAL>
+class KContainerT
 {
+public:
+  using Real               = REAL;
+  using FullPrecReal       = QMCTraits::FullPrecRealType;
+  using PositionFull       = QMCTraits::QTFull::PosType;
+  static constexpr int DIM = OHMMS_DIM;
+  using AppPosition        = typename QMCTraits::PosType;
+  using Position           = typename QMCTypes<Real, DIM>::PosType;
```

Comment:
With this change, there are three precision in this class.
`REAL`, `QMCTraits::FullPrecRealType` and `QMCTraits::PosType` (effectively `QMCTraits::RealType`)
This seems defeating the purpose of templates.